clients/src/main/java/org/apache/kafka/common/requests/CreatePartitionsResponse.java

```diff
@@ -52,10 +51,9 @@ protected Struct toStruct(short version) {
     @Override
     public Map<Errors, Integer> errorCounts() {
         Map<Errors, Integer> counts = new HashMap<>();
```

Comment:
Could we leverage the new method ```errorCounts(Stream<Errors>)```? For example:
```
return errorCounts(data.results().stream().map(result -> Errors.forCode(result.errorCode())));
```

Reply:
@chia7712 what I've tried to do in this PR so far is:

* Change `for` stmt  + updateErrorCounts to use `forEach` consistently
* Change calls to `errorCounts(Collection)` to `errorCounts(Stream)`

I've not tried to change all code to use either `forEach` or `errorCounts(Stream)`. Obviously we could do that, but @ijuma seems happy enough with continuing to have these two ways to do it.